src/components/EditorWidgets/Date/DateControl.js

```diff
@@ -20,9 +20,14 @@ export default class DateControl extends React.Component {
     includeTime: PropTypes.bool,
   };
 
-  componentDidMount() {
-    const { includeTime, value, field, onChange } = this.props;
+  constructor(props) {
+    super(props);
+    const { field, includeTime } = this.props;
     this.format = field.get('format') || (includeTime ? DEFAULT_DATETIME_FORMAT : DEFAULT_DATE_FORMAT);
+  }
+
```

Comment:
@tech4him1 any reason in constructor rather than in componentWillMount

I would have to test, but I would think the format could change from one collection to the next right?

Reply:
The format can change between collections.

Can you explain when `componentWillMount` should be used versus `constructor`?

Reply:
I may have made a bad assumption here. I thought we were talking about state changes 😛

Reply:
My assumption was incorrect, sorry for not looking closer first.